### PR TITLE
(morpho-v1) Fix DLL after max iterations

### PR DESCRIPTION
**Pull request by QGarchery**
Test and fix max iterations behavior: the DLL should insert at the end when the maximum number iterations is reached, regardless of the values stored in the DLL

### CODE REVIEW

**Review by MerlinEgalite — APPROVED** [review]
LGTM! Can you do the same changes on the main branch please?

**Comment by QGarchery** [comment]
> LGTM! Can you do the same changes on the main branch please?

It is already done in https://github.com/morpho-dao/morpho-data-structures/pull/97, this PR is ready for review

**Review by pakim249CAL — APPROVED** [review]

File: contracts/DoubleLinkedList.sol [hunk 1]
```diff
@@ -115,32 +119,34 @@ library DoubleLinkedList {
...
-                _list.accounts[_id] = Account(address(0), address(0), _value);
+                _list.accounts[_id] = Account({prev: address(0), next: address(0), value: _value});
                 _list.head = _id;
                 _list.tail = _id;
             }
             // Account is not the new head.
             else {
-                _list.accounts[_id] = Account(_list.tail, address(0), _value);
-                _list.accounts[_list.tail].next = _id;
+                address tail = _list.tail;
```
Comment (MathisGD) [inline contracts/DoubleLinkedList.sol:147]: This variable increases the gas cost for me 👀

File: contracts/DoubleLinkedList.sol [hunk 2]
```diff
@@ -115,32 +119,34 @@ library DoubleLinkedList {
...
-                _list.accounts[_id] = Account(address(0), next, _value);
+                _list.accounts[_id] = Account({prev: address(0), next: next, value: _value});
                 _list.head = _id;
                 _list.accounts[next].prev = _id;
             }
             // Account is not the new head.
             else {
-                _list.accounts[_id] = Account(_list.accounts[next].prev, next, _value);
-                _list.accounts[_list.accounts[next].prev].next = _id;
+                address prev = _list.accounts[next].prev;
```
Comment (MathisGD) [inline contracts/DoubleLinkedList.sol:131]: This variable does not decrease the gas cost for me 👀

**Review by MathisGD — APPROVED** [review]
Optimizations, otherwise LGTM

**Review by Rubilmax — APPROVED** [review]

File: contracts/DoubleLinkedList.sol (hunk 1 above)
Comment (QGarchery) [reply contracts/DoubleLinkedList.sol:147]: True, let's keep that in mind (won't do for now as it is changing the bytecode)

File: contracts/DoubleLinkedList.sol (hunk 2 above)
Comment (QGarchery) [reply contracts/DoubleLinkedList.sol:131]: same as above

**Pull request merged**